src/test.js

```diff
@@ -706,33 +753,39 @@ extend( test, {
 
 		newTest.queue();
 	},
-	skip: function skip( testName ) {
-		if ( focused || config.currentModule.ignored ) {
-			return;
-		}
-
-		const test = new Test( {
-			testName: testName,
-			skip: true
-		} );
-
-		test.queue();
+	skip: function( testName ) {
+		skip( testName, undefined );
+	},
+	only: function( testName, callback ) {
+		only( testName, undefined, callback );
 	},
-	only: function only( testName, callback ) {
+	each: function( testName, data, callback ) {
 		if ( config.currentModule.ignored ) {
 			return;
 		}
-		if ( !focused ) {
-			config.queue.length = 0;
-			focused = true;
-		}
 
-		const newTest = new Test( {
-			testName: testName,
-			callback: callback
+		runEach( data, ( datum, i ) => {
+			const newTest = new Test( {
+				testName: `${i} ${testName}`,
+				callback: callback,
+				params: datum
+			} );
+			newTest.queue();
 		} );
+	}
+} );
 
-		newTest.queue();
+
+extend( test.each, {
+	skip: function( testName, data ) {
+		runEach( data, ( datum, i ) => {
+			skip( `${i} ${testName}`, datum );
```

Comment:
Do we really need to pass `datum` through here? I think leaving it off (and becoming `undefined`) would be just as well, like how `skip` already doesn't need the callback.

Reply:
Ah we do not. Removed it.